Frameworks/ImageIO/CGImageDestination.mm

```diff
@@ -15,11 +16,867 @@
 //******************************************************************************
 
 #import <ImageIO/CGImageDestination.h>
+#import <ImageIO/CGImageDestinationInternal.h>
 #import <StubReturn.h>
+#include <windows.h>
+#include <string>
 
+static const wchar_t* TAG = L"CGImageDestination"; 
 const CFStringRef kCGImageDestinationLossyCompressionQuality = static_cast<CFStringRef>(@"kCGImageDestinationLossyCompressionQuality");
 const CFStringRef kCGImageDestinationBackgroundColor = static_cast<CFStringRef>(@"kCGImageDestinationBackgroundColor");
 
+const CFStringRef kUTTypeJPEG = static_cast<const CFStringRef>(@"public.jpeg");
+const CFStringRef kUTTypeTIFF = static_cast<const CFStringRef>(@"public.tiff");
+const CFStringRef kUTTypeGIF = static_cast<const CFStringRef>(@"com.compuserve.gif");
+const CFStringRef kUTTypePNG = static_cast<const CFStringRef>(@"public.png");
+const CFStringRef kUTTypeBMP = static_cast<const CFStringRef>(@"com.microsoft.bmp");
+
+const unsigned int c_minutesPerDegree = 60;
+const unsigned int c_secondsPerMinute = 60;
+
+enum imageTypes { typeJPEG,
+                  typeTIFF,
+                  typeGIF,
+                  typePNG,
+                  typeBMP,
+                  typeUnknown };
+
+enum destinationTypes { toData,
+                        toURL,
+                        toConsumer };
+
+@implementation ImageDestination
+
+- (instancetype)initToDestination:(size_t)frames
+                             type:(CFStringRef)type
+                             data:(CFMutableDataRef)data
+                              url:(CFURLRef)url {
+    if (self = [super init]) {
+        RETURN_NULL_IF(url && ![(NSURL*)url isFileURL]);
+        RETURN_NULL_IF(!data && !url);
+
+        self.maxCount = frames;
+
+        MULTI_QI imageQueryInterface = {0};
+        static const GUID IID_IWICImagingFactory = {0xec5ec8a9,0xc395,0x4314,0x9c,0x77,0x54,0xd7,0xa9,0x35,0xff,0x70};
+        imageQueryInterface.pIID = &IID_IWICImagingFactory;
+        RETURN_NULL_IF_FAILED(
+            CoCreateInstanceFromApp(CLSID_WICImagingFactory, nullptr, CLSCTX_INPROC_SERVER, nullptr, 1, &imageQueryInterface));
+        
+        _idFactory = (IWICImagingFactory*)imageQueryInterface.pItf;
+        RETURN_NULL_IF_FAILED(_idFactory->CreateStream(&_idStream));
+
+        if (CFStringCompare(type, kUTTypeJPEG, NULL) == kCFCompareEqualTo) {
```

Comment:
I replaced all the NULLs with nullptr except for CGStringCompare, which doesn't compile with nullptr or nil. I'm not sure why, but I left a comment there saying that's why it's being left as NULL.